parsl/executors/high_throughput/executor.py

```diff
@@ -567,11 +568,11 @@ def submit(self, func, resource_specification, *args, **kwargs):
         Returns:
               Future
         """
-        if resource_specification:
-            logger.error("Ignoring the resource specification. "
-                         "Parsl resource specification is not supported in HighThroughput Executor. "
-                         "Please check WorkQueueExecutor if resource specification is needed.")
-            raise UnsupportedFeatureError('resource specification', 'HighThroughput Executor', 'WorkQueue Executor')
+        if 'app_type' in call_specs and len(call_specs) > 1:
+            logger.error("Ignoring the call specification. "
+                         "Parsl call specification is not supported in HighThroughput Executor. "
+                         "Please check WorkQueueExecutor or TaskVineExecutor if call specification is needed.")
+            raise UnsupportedFeatureError('call specification', 'HighThroughput Executor', 'WorkQueue Executor')
```

Comment:
this exception is a bit icky: it's meant to direct you to use another executor - but it doesn't redirect you to either WQ or TaskVine in the exception object.

That's not too concerning but it's a bit ugly... I think the right thing here is that the UnsupportedFeatureError shouldn't be pointing people to WQ at all: in a modular, open expanding world, the High Throughput Executor code shouldn't be expected to know about other executors at all...

Reply:
I agree. UnsupportedFeatureError is now agnostic to target executors if needs be.